std/assembly/array.ts

```diff
@@ -153,6 +153,29 @@ export class Array<T> {
     return newLength;
   }
 
+  concat(items: Array<T>): Array<T> {
+    assert(this != null);
+
+    var thisLen: isize = this.length_;
+    var otherLen = (items == null) ? 0 : items.length_;
+    var outLen =  thisLen + otherLen;
+
+    var out: Array<T> = new Array<T>(outLen);
+    memory.copy(changetype<usize>(out.buffer_) + HEADER_SIZE,
```

Comment:
I think make sense add non-zero length checking also for current array:
```ts
if (thisLen) {
   memory.copy(...)
}
```
And add tests for:
- [].concat(arr)
- arr.concat([])

Reply:
@MaxGraey I had enhanced the code as you suggested. 
But test case [`assert(source.length == 3);`](https://github.com/LiaoPeng/assemblyscript/blob/d97fb8a21aaac818ab04d128ae7430be4eeab1a8/tests/compiler/std/array.ts#L141) was failed. But I can't find the bugs, can you help me to solve  the issue?

Reply:
Tests should looks like this:
```ts
var source: i32[] = [];
assert(source.length == 0);
out = source.concat(arr);
assert(out.length == 3);
assert(source.length == 0);
```